File: dataset/Vanilla - Copy/GEMINI_gemini-2.5-pro/Task168.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

/**
 * @brief Tweaks letters by one forward (+1) or backwards (-1) according to an array.
 * 
 * @param str The input string of lowercase letters.
 * @param tweaks An array of integers (0, 1, or -1).
 * @param len The length of the string and the tweaks array.
 * @return A new dynamically allocated string with the tweaked letters. The caller must free this string.
 */
char* tweakLetters(const char* str, const int* tweaks, size_t len) {
    char* result = (char*)malloc(len + 1);
    if (result == NULL) {
        perror("Failed to allocate memory");
        return NULL;
    }

    for (size_t i = 0; i < len; ++i) {
        char c = str[i];
        int tweak = tweaks[i];
        char newChar = c + tweak;

        if (newChar > 'z') {
            newChar = 'a';
        } else if (newChar < 'a') {
            newChar = 'z';
        }
        result[i] = newChar;
    }
    result[len] = '\0';
    return result;
}

void run_test(const char* str, const int* tweaks, size_t len) {
    char* result = tweakLetters(str, tweaks, len);
    if (result) {
        printf("%s\n", result);
        free(result);
    }
}

int main() {
    // Test Case 1
    int tweaks1[] = {0, 1, -1, 0, -1};
    run_test("apple", tweaks1, sizeof(tweaks1)/sizeof(int));

    // Test Case 2
    int tweaks2[] = {0, 0, 0, -1};
    run_test("many", tweaks2, sizeof(tweaks2)/sizeof(int));

    // Test Case 3
    int tweaks3[] = {1, 1, 1, 1, 1};
    run_test("rhino", tweaks3, sizeof(tweaks3)/sizeof(int));

    // Test Case 4
    int tweaks4[] = {1, -1, 1, -1, 1};
    run_test("zebra", tweaks4, sizeof(tweaks4)/sizeof(int));

    // Test Case 5
    int tweaks5[] = {-1, 0, 1};
    run_test("abc", tweaks5, sizeof(tweaks5)/sizeof(int));

    return 0;
}

File: dataset/Vanilla/GEMINI_gemini-2.5-pro/Task168.py
def tweak_letters(s, tweaks):
    """
    Tweaks letters by one forward (+1) or backwards (-1) according to a list.
    
    :param s: The input string of lowercase letters.
    :param tweaks: A list of integers (0, 1, or -1).
    :return: The tweaked string.
    """
    if len(s) != len(tweaks):
        return "" # Or raise an error
        
    result = []
    for i in range(len(s)):
        char_code = ord(s[i]) + tweaks[i]
        
        if char_code > ord('z'):
            result.append('a')
        elif char_code < ord('a'):
            result.append('z')
        else:
            result.append(chr(char_code))
            
    return "".join(result)

if __name__ == '__main__':
    # Test Case 1
    print(tweak_letters("apple", [0, 1, -1, 0, -1]))
    
    # Test Case 2
    print(tweak_letters("many", [0, 0, 0, -1]))
    
    # Test Case 3
    print(tweak_letters("rhino", [1, 1, 1, 1, 1]))
    
    # Test Case 4
    print(tweak_letters("zebra", [1, -1, 1, -1, 1]))
    
    # Test Case 5
    print(tweak_letters("abc", [-1, 0, 1]))